Tools/cases_generator/generate_cases.py

```diff
@@ -291,24 +320,38 @@ def parse(self) -> None:
             raise psr.make_syntax_error(
                 f"Couldn't find {BEGIN_MARKER!r} in {psr.filename}"
             )
+        start = psr.getpos()
 
-        # Parse until end marker
+        # Find end marker, then delete everything after it
+        while tkn := psr.next(raw=True):
+            if tkn.text == END_MARKER:
+                break
+        del psr.tokens[psr.getpos() - 1 :]
+
+        # Parse from start
+        psr.setpos(start)
```

Comment:
This tactic makes it possible to write the main parsing loop as `while thing := psr.definition():`.